tutorials/tutorial-two-go-stream.md

```diff
@@ -0,0 +1,379 @@
+Make sure to follow [the setup steps](/tutorials/tutorial-one-go-stream#setup) from the first tutorial.
+
+An executable version of this tutorial can be found in the [RabbitMQ tutorials repository](https://github.com/rabbitmq/rabbitmq-tutorials/blob/main/go-stream/).
+The sending program is called `offset_tracking_send.go` and the receiving program is called `offset_tracking_receive.go`.
+The tutorial focuses on the usage of the client library, so the final code in the repository should be used to create the scaffolding of the files (e.g. imports, main function, etc).
+
+### Sending
+
+The sending program starts by creating the environment and declaring the stream:
+
+```go
+env, _ := stream.NewEnvironment(
+    stream.NewEnvironmentOptions().
+        SetHost("localhost").
+        SetPort(5552).
+        SetUser("guest").
+        SetPassword("guest"))
+
+streamName := "stream-offset-tracking-go"
+env.DeclareStream(streamName,
+    &stream.StreamOptions{
+        MaxLengthBytes: stream.ByteCapacity{}.GB(2),
+    },
+)
+```
+
+Note the error handling code is omitted for brevity.
+
+The program then creates a producer and publishes 100 messages.
+The body value of the last message is set to `marker`; this is a marker for the consumer to stop consuming.
+
+The program uses a `handlePublishConfirm` function and a channel to make sure all messages make it to the broker before exiting.
+Let's skip this part for now and see the bulk of the sending first:
+
+```java
+producer, _ := env.NewProducer(streamName, stream.NewProducerOptions())
+
+messageCount := 100
+ch := make(chan bool)
+chPublishConfirm := producer.NotifyPublishConfirmation()
+handlePublishConfirm(chPublishConfirm, messageCount, ch)
+
+fmt.Printf("Publishing %d messages\n", messageCount)
+for i := 0; i < messageCount; i++ {
+    var body string
+    if i == messageCount-1 {
+        body = "marker"
+    } else {
+        body = "hello"
+    }
+    producer.Send(amqp.NewMessage([]byte(body)))
+}
+_ = <-ch
+fmt.Println("Messages confirmed")
+
+producer.Close()
+```
+
+The sending program uses 2 channels and a Go routine to move on and exit when the broker confirmed the messages.
+Let's focus on this part.
+
+The producer `NotifyPublishConfirmation` function returns the first channel.
+The client library sends the broker confirmations on this channel and a routine declared in `handlePublishConfirm` receives these confirmations:
+
+```go
+messageCount := 100
+chPublishConfirm := producer.NotifyPublishConfirmation()
+ch := make(chan bool)
+handlePublishConfirm(chPublishConfirm, messageCount, ch)
+```
+
+The routine deals with the messages and sends `true` to the second channel when the expected number of confirmations is reached:
+
+```go
+func handlePublishConfirm(confirms stream.ChannelPublishConfirm, messageCount int, ch chan bool) {
+	go func() {
+		confirmedCount := 0
+		for confirmed := range confirms {
+			for _, _ = range confirmed {
```

Comment:
Here, we count the confirmations,  but:
```golang
	for _, msg := range confirmed {
		 if msg.IsConfirmed() {
		
```

With `.IsConfirmed()`, you are sure that the server has confirmed the message. 

not sure we have to deal with `.IsConfirmed()` here or maybe directly to another tutorial.